x/crosschain/keeper/evm_hooks.go

```diff
@@ -257,28 +286,33 @@ func (k Keeper) ParseZRC20WithdrawalEvent(ctx sdk.Context, log ethtypes.Log) (*z
 	if err != nil {
 		return nil, err
 	}
```

Comment:
i would consider moving the foreign coin ZRC20 address check to here; 
this is minor though: consider a situation that I deployed a contract that happens to emit the same Withdrawal event; 
Now my tx would fail if it does not pass the validation, which is weird.

Reply:
I'm not sure I understand. Is there a reason we would want to process an event emitted by an external contract?
We continue if the coin check fails,so it will just skip